crates/sui-upload-display/src/tests.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{AccumulatedDisplayData, DisplayEntry};
+use anyhow::Result;
+use std::path::PathBuf;
+use tempfile::tempdir;
+
+fn create_dummy_display_entry(idx: u8) -> DisplayEntry {
+    DisplayEntry {
+        object_type: vec![0x01, 0x02, 0x03, idx],
+        display_id: vec![0x04, 0x05, 0x06, idx],
+        display_version: 1,
+        display: vec![0x07, 0x08, 0x09, idx],
+    }
+}
+
+fn create_dummy_entries(count: u8) -> Vec<DisplayEntry> {
+    (0..count).map(create_dummy_display_entry).collect()
+}
+
+fn write_entries_to_csv(entries: &[DisplayEntry], path: &PathBuf) -> Result<()> {
+    let mut writer = csv::WriterBuilder::new()
+        .has_headers(true)
+        .from_path(path)?;
+
+    writer.write_record(["object_type", "display_id", "display_version", "display"])?;
+
+    for entry in entries {
+        let object_type_hex = hex::encode(&entry.object_type);
+        let display_id_hex = hex::encode(&entry.display_id);
+        let display_hex = hex::encode(&entry.display);
+
+        writer.write_record([
+            &format!("\\x{}", object_type_hex),
+            &format!("\\x{}", display_id_hex),
+            &entry.display_version.to_string(),
+            &format!("\\x{}", display_hex),
+        ])?;
+    }
+
+    writer.flush()?;
+    Ok(())
+}
+
+fn extract_entries(data: &AccumulatedDisplayData) -> Vec<DisplayEntry> {
+    data.displays.values().cloned().collect()
+}
+
+fn compare_entries(a: &[DisplayEntry], b: &[DisplayEntry]) -> bool {
```

Comment:
I might be missing some nuance here, but this function looks like it's performing a set comparison on two vectors -- is that right? (A doc comment explaining what this function was doing would also have helped make this clearer) If so, why not `collect` the values into sets and compare those?:


```rust
BTreeSet::from_iter(a.iter()) == BTreeSet::from_iter(b.iter())
```

?